File: leetcode/Medium/1823. Find the Winner of the Circular Game/1823.cpp
class Solution {
public:
    int helper(int now, int k, vector<int> v){
        if(v.size() == 1) return v[0];
        int idx = (now + k) % v.size();
        v.erase(v.begin() + idx);
        return helper(idx, k, v);
    }
    int findTheWinner(int n, int k) {
        vector<int> v;
        for(int i=1; i<=n; i++)
            v.push_back(i);

        return helper(0, k-1, v);
    }
};

// TC: O(n + k)
// SC: O(n)

File: leetcode/Medium/1823. Find the Winner of the Circular Game/1823.java
class Solution {
    
    public int findTheWinner(int n, int k) {
        List<Integer> list = new ArrayList<>();
        for(int i=1; i<=n; i++){
            list.add(i);
        }
        return helper(0, k-1, list);
    }

    private int helper(int loc, int k, List<Integer> list){
        if(list.size() == 1) return list.get(0);
        int idx = (loc + k) % list.size();
        list.remove(idx);
        return helper(idx, k, list);
    }
}

// TC: O(n + k)
// SC: O(n)
